fastavro/schema.py

```diff
@@ -67,7 +67,7 @@ def extract_named_schemas_into_repo(schema, repo, transformer, parent_ns=None):
 
     namespace, name = schema_name(schema, parent_ns)
 
-    if name and (name not in repo):
```

Comment:
I can't find anything in the spec. Perhaps @pkoch can shed some light as he was the one who originally added it. If I had to guess it was probably just some micro optimization about not reading/writing to the cache dict if the schema was already in there. But that's precisely the problem. A second record with the same name could come along and when reads/writes are done, the first schema that was added is used. The test case I added fails for me under the current master.